Fix cmake install on dev branch

jngrad opened the pull request:
Porting #3228 to the dev branch

Description of changes:
- fix the CMake build system to include libraries in the list of installed files and test it in a dedicated CI job
- fix the versioning of Cython .so shared objects
- simplify the CMake tests

junghans committed:
cmake: install missing targets

junghans committed:
cmake: remove SOVERSION

junghans committed:
cmake: fix permission of python module install

jngrad committed:
Document CMake logic

jngrad committed:
Fix version nomenclature in Sphinx docs

jngrad committed:
Deprecate DESTDIR install method

Installing ESPResSo with `make install DESTDIR="/path/to/es"` will
install files in the wrong folder and set an incorrect runtime path
in Cython shared objects, causing import errors. See full report:
https://github.com/espressomd/espresso/pull/3228#issuecomment-541422452

jngrad committed:
Fix cluster_analysis module

When installing ESPResSo, the C++ shared objects and Cython shared
objects are stored in the same folder. Since they have the same .so
extension, when importing espressomd.cluster_analysis, the file
cluster_analysis.py is loaded and the file cluster_analysis.so is
automatically considered a Cython shared object, and Python fails
to find the Cython-specific symbol PyInit_cluster_analysis. This
is remediated by giving a different name to the C++ shared object.

jngrad committed:
Add job to check the installed version of ESPResSo

Install the files in a different directory and check the python and
tutorials/samples tests can still run with the installed files.
Please note that pypresso will experience a major slowdown in CI
when located outside the build directory, hence the 3h timeout.
This job is meant to be run only during releases and in PRs that
add new core modules.

jngrad committed:
Skip tutorial ferrofluid part 3 (timeout)

Takes more than 1 hour to run in CI when pypresso is located
outside the build directory.

jngrad committed:
Remove redundant PYTHONPATH

jngrad committed:
Simplify CMake test logic

The installed files only exist if the Python bindings are enabled.

KaiSzuttor commented on testsuite/cmake/CMakeLists.txt:
```diff
@@ -12,29 +12,17 @@ function(CMAKE_TEST)
-  cmake_test(FILE test_install_prefix.sh DEPENDENCIES BashUnitTests.sh)
+  cmake_test(FILE test_install.sh DEPENDENCIES BashUnitTests.sh)
 endif()
+set(DESTDIR "${CMAKE_CURRENT_BINARY_DIR}/install")
```
do we need it?

KaiSzuttor reviewed (approved):
besides comment LGTM

jngrad committed:
Remove unused DESTDIR variable

jngrad committed:
Cannot test python3 in CI

The clang:6.0 job fails due to undefined ASAN symbols.
The logic to fix that error is in the pypresso wrapper.

jngrad commented:
bors r=KaiSzuttor
